src/irmin-pack/unix/mapping_file.ml

```diff
@@ -328,6 +328,35 @@ let calculate_extents_oc ~(src_is_sorted : unit) ~(src : int_bigarray)
   in
   dst_off
 
+let rev_inplace (src : int_bigarray) : unit =
+  let src_sz = BigArr1.dim src in
+  let _ =
+    assert (src_sz >= 3);
+    assert (src_sz mod 3 = 0)
+  in
+  let rec rev i j =
+    if i < j then (
+      let ioff, ilen = (src.{i}, src.{i + 2}) in
+      let joff, jlen = (src.{j}, src.{j + 2}) in
+      src.{i} <- joff;
+      src.{i + 2} <- jlen;
+      src.{j} <- ioff;
+      src.{j + 2} <- ilen;
+      rev (i + 3) (j - 3))
+  in
+  rev 0 (src_sz - 3)
+
+let set_prefix_offsets src =
+  let src_sz = BigArr1.dim src in
+  let rec go i poff =
+    if i < src_sz then (
+      assert (poff >= 0);
```

Comment:
maybe this assert is not necessary?

Reply:
@art-w any thoughts on this? (I tend to agree that it is not needed)